acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+>>> date = '2016-04-01'
+>>> detector='WFC'
+>>> filt = 'F435W'
+>>> q = acszpt.Query(date=date, detector=detector, filter=filt)
+>>> zpt_table = q.fetch()
+>>> print(zpt_table)
+FILTER PHOTPLAM        PHOTFLAM         STmag  VEGAmag  ABmag
+       Angstrom erg / (Angstrom cm2 s) mag(ST)   mag   mag(AB)
+------ -------- ---------------------- ------- ------- -------
+ F435W   4329.2              3.148e-19  25.155  25.763  25.665
+"""
+
+
+
+
+from astropy.table import Table
+import astropy.units as u
+from collections import OrderedDict
+import datetime as dt
+import requests
+from bs4 import BeautifulSoup
+
+
+__taskname__ = "acszpt"
+__version__ = "1.0"
+__author__ = "Nathan Miles"
+__vdate__ = "22-Jan-2019"
+__all__ = ['acszpt']
+
+
+
+class Query(object):
+    """ Class used to interface with the ACS Zeropoints Calculator API.
+
+    Parameters
+    ----------
+    date : str
+        Input date in the following ISO format, YYYY-MM-DD
+    detector : str
+        One of the three channels on ACS: HRC, SBC, or WFC
+    filt : str, optional
+        One of valid filters for the chosen detector
+            * HRC:
+                 F220W, F250W, F330W,
+                 F344N, F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+
+            * WFC:
+                 F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+            * SBC:
+                F115LP, F122M, F125LP,
+                F140LP, F150LP, F165LP
+
+    Attributes
+    ----------
+    date : str
+        **Attribute** for storing the user specified date.
+    detector : str
+        **Attribute** for storing the user specified detector.
+    filt : str, None
+        **Attribute** for storing the user specified filter (if supplied).
+    zpt_table : astropy.table.Table
+        **Attribute** for storing the results returned by the ACS Zeropoint Calculator
+    """
+
+    def __init__(self, date, detector, filt=None):
+
+        # Set the public attributes
+        self.date = date
+        self.detector = detector.upper()
+        self.filt = filt
+
+        self.valid_filters = {
+            'WFC': ['F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'HRC': ['F220W', 'F250W', 'F330W', 'F344N',
+                    'F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'SBC': ['F115LP', 'F122M', 'F125LP',
+                   'F140LP', 'F150LP', 'F165LP']
+        }
+        self.zpt_table = None
+
+        # Set the private attributes
+        if filt is None:
+            self._url = 'https://acszeropoints.stsci.edu/results_all/?date={}&' \
```

Comment:
Try to avoid `\` continuation if possible. Use parenthesis instead.

Reply:
Just for my own edification, is there a distinct advantage to using parenthesis vs `\` or is this just a stylistic request?